packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -194,24 +220,56 @@ export default util.createRule<Options, MessageId>({
       if (
         !isPromiseLike(checker, parserServices.esTreeNodeToTSNodeMap.get(node))
       ) {
-        return false;
+        return { isUnhandled: false };
       }
 
       if (node.type === AST_NODE_TYPES.CallExpression) {
         // If the outer expression is a call, it must be either a `.then()` or
         // `.catch()` that handles the promise.
-        return (
-          !isPromiseCatchCallWithHandler(node) &&
-          !isPromiseThenCallWithRejectionHandler(node) &&
-          !isPromiseFinallyCallWithHandler(node)
-        );
+        //
+        // TODO (Kirk) - This is not what the code does. It also allows a .finally
+        // so long as it has a handler.
```

Comment:
Oh, whoops, I'll remove that.

I had made that note to myself while contemplating whether to include validation to the `.finally` handler, since I noticed that the comment there is out of sync with the code. (i.e. "if the outer expression is a call, it must be either a .then or .catch that handles the promise" is incomplete, since the code also permits .finally).